### PR TITLE
feat: add Rust parser

### PR DESCRIPTION
**Pull request by yijunyu**
Adapted `DFG_csharp` to `DFG_rust` in order to parse Rust code using the `tree-sitter-rust` parser.

### CODE REVIEW

File: use.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# prediction = "void add (int a ,int b ) {\n return a + b;}"
+# reference = "void sum ( int first , int second ) {\n return second + first;}"
+# result = calc_codebleu([reference], [prediction], lang="c", weights=(0.25, 0.25, 0.25, 0.25), tokenizer=None)
+# print(result)
+
+prediction = "fn add ( a , b )->i8 {\n a + b}"
+reference = "fn sum ( first , second )->i8 {\n second + first}"
+result = calc_codebleu([reference], [prediction], lang="rust", weights=(0.25, 0.25, 0.25, 0.25), tokenizer=None)
+print(result)
+
```
Comment (k4black) [inline use.py:25]: Could you please delete this file?

File: codebleu/codebleu.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from . import bleu, dataflow_match, syntax_match, weighted_ngram_match
 
 PACKAGE_DIR = Path(__file__).parent
-AVAILABLE_LANGS = ["java", "javascript", "c_sharp", "php", "c", "cpp", "python", "go", "ruby"]  # keywords available
+AVAILABLE_LANGS = ["java", "javascript", "c_sharp", "php", "c", "cpp", "python", "go", "ruby", "rust"]  # keywords available
```
Comment (k4black) [inline codebleu/codebleu.py:10]: Could you also add "rust" to
1. README.md
2. evaluate_app/README.md
2. evaluate_app/codebleu.py - description

File: codebleu/parser/DFG.py [hunk 3]
```diff
@@ -1213,3 +1213,176 @@ def DFG_javascript(root_node, index_to_code, states):
                 DFG += temp
 
         return sorted(DFG, key=lambda x: x[1]), states
+
+
+def DFG_rust(root_node, index_to_code, states):
```
Comment (k4black) [inline codebleu/parser/DFG.py:1218]: Please check pipeline failed, you need to apply `black` formatter (`python -m black .`)
Also pipeline if checking isort/ruff/mypy

**Review by k4black — CHANGES_REQUESTED** [review]
Hi! Thank you a lot! Great efforts  
Could you please fix couple of things I mention in the review comment.

Additionally, could you please tests for rust in `tests/test_codebleu.py` file `test_simple_cases_work_for_all_langs` test case to be sure it is working on all devices.

**Comment by yijunyu** [comment]
Could you take a look at the format fixes? Thanks -- Yijun

**Comment by k4black** [comment]
@yijunyu Thanks for fixes! 
Regarding formatting im happy if it passes the lining pipeline (it does) thanks! 

Could you please fix other issues i mentioned? Add README info about rust, remove extra file, add test case?
Or i can make it myself, please let me know!

**Comment by yijunyu** [comment]
Please do so. 

I am a bit tied up recently. 

The Readme won't be significantly different to those for existing languages. If you need help I can start with a similar document.

I have a test case which I may add to the report for this, which fine shall I put it ? Or maybe I just paste it here.

**Pull request merged**